### DISCUSSION TITLE
AR failing to place the model on Android

### DISCUSSION THREAD

**Discussion by Giantbean** [google/model-viewer#4854] [Q&A] [answered]
I am testing model viewer and I have made a simple page to view a model. This works in the web page to view and rotate the model and with an iPad I can start AR and place the model but on android the AR camera starts however the model is not shown and cannot be placed.

```
<div>
        <model-viewer src="assets\models\anyModel.gltf"
        poster="assets/images/anImageOfTheModel.png"
        alt="A 3d model"     
        ar 
        ar-modes="webxr scene-viewer quick-look"
        tone-mapping="neutral"  
        shadow-intensity="1.15" shadow-softness="0.6"
        camera-orbit="-94.43deg 84.34deg 204.1m" field-of-view="30deg" 
        min-camera-orbit="auto 60deg auto" max-camera-orbit="auto 85deg auto"
        camera-controls touch-action="pan-y">
        </model-viewer>
    </div>
```

    

    I have tried a few variations of ar-modes mapping and so on but nothing has worked on android. 
    One+ Nord Android V11 
    
    I also tried to make it work with older browsers

    
```
   <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial=scale=1.0">
    <!-- Loads <model-viewer> for modern browsers: -->
    <script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>
    <!-- Loads <model-viewer> for old browsers like IE11: -->
    <script nomodule src="https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js"></script>
    <title>ModelTest</title>
  </head>
```

    And have a simple style for layout

    
```
  <style>
    model-viewer {width: 400px; height: 600px; margin:0, auto;}
  </style>
```
 
 Thanks.

**Comment by elalish** [comment, marked as answer]
Wild guess is you have the wrong units in your GLB. All glTF files are in meters, and the fact that you set your camera orbit to 204.1 m tells me your model is probably way too large. You won't notice until you go into AR, because then it's always shown at true scale and it's probably towering over you or something.

**Reply by elalish** [reply]
A quick fix is to try `scale`: https://modelviewer.dev/docs/index.html#entrydocs-scenegraph-attributes-scale

**Reply by Giantbean** [reply]
Follow up question:
I'm not quite understanding that documentation. How do I set the scale when it launches AR on android? It works in web view and on iOS but I'm not sure what to do in the code? 
Something like modelViewerTransform.scale = `${0.25} ${0.25} ${0.25}`;}; ? Or is there just a Scale = "0.25" attribute that I am missing? Or do I need to adjust the mesh? 
It's an architectural model to be placed on site hence the large scale.

Edit:
Trying scale = "${0.5} ${0.5} ${0.5}"
Sort of worked but I don't see how to place the model it just picks a point and then I can pinch to scale the mesh and rotate it with a swipe but there is no preview to place the mesh in the proper location like one sees on android. Is that just a limitation of webXR compared to quick-look?

**Reply by elalish** [reply]
This is a different question, but yes you can also drag the model around to reposition it. However, we also appear to have a regression here I'm currently looking into, so if it behaves a little oddly, that's probably why. And yes, we place the model initially automatically so the experience doesn't seem broken if you can't manage to get the floor to scan (or there is no visible floor).
